Modules/_json.c

```diff
@@ -1251,6 +1251,25 @@ encoder_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     return (PyObject *)s;
 }
 
+static PyObject *
+_create_newline_indent(PyObject *indent, Py_ssize_t indent_level)
+{
+    PyObject *current_indent = PySequence_Repeat(indent, indent_level);
+    if (current_indent == NULL) {
+        return NULL;
+    }
+    PyObject *start = PyUnicode_FromOrdinal('\n');
+    if (start == NULL) {
+        Py_DECREF(current_indent);
+        return NULL;
+    }
+
+    PyObject *newline_indent = PyUnicode_Concat(start, current_indent);
```

Comment:
Why don't we use the previous `newline_indent` here? (Like I did in my own json encoder).
`newline_indent` is intialised at `'\n'`, and each recursive call we concat `indent`.

```diff
-    PyObject *newline_indent = PyUnicode_Concat(start, current_indent);
+    PyObject *new_newline_indent = PyUnicode_Concat(newline_indent, indent);
```

Reply:
Sorry, my bad, this is only used for the initial indent level. (I made a wrong assumption when seeing this function.

But why do we need it? The C encoder is always called with `_current_indent_level` set to 0.

Reply:
I think you are right we don't need it. It can currently be set via `_json.make_encoder` and `_json` is not part of the public interface so the argument could be removed there. 

But again, I want to be conservative in how much this PR changes. If one the core devs thinks we should remove the argument I will remove it (and then we can indeed simplify `_create_newline_indent `)